CMakeLists.txt

```diff
@@ -66,76 +75,98 @@ install(DIRECTORY include/${PROJECT_NAME} DESTINATION include)
+            DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}
+        )
+
+        install(DIRECTORY
+            config
+            launch
+            DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
+        )
+
+        if(CATKIN_ENABLE_TESTING)
+            ## Add gtest based cpp test target and link libraries
+            catkin_add_gtest(steering_functions_test test/steering_functions_test.cpp)
+            if(TARGET steering_functions_test)
+                target_link_libraries(steering_functions_test ${catkin_LIBRARIES} ${PROJECT_NAME})
+                target_include_directories(steering_functions_test PRIVATE ${catkin_INCLUDE_DIRS})
+            endif()
+
+            catkin_add_gtest(fresnel_test test/fresnel_test.cpp)
+            if(TARGET fresnel_test)
+                target_link_libraries(fresnel_test ${catkin_LIBRARIES} ${PROJECT_NAME})
+                target_include_directories(fresnel_test PRIVATE ${catkin_INCLUDE_DIRS})
+            endif()
+
+            catkin_add_gtest(timing_test test/timing_test.cpp)
+            if(TARGET timing_test)
+                target_link_libraries(timing_test ${catkin_LIBRARIES} ${PROJECT_NAME})
+                target_include_directories(timing_test PRIVATE ${catkin_INCLUDE_DIRS})
+            endif()
+
+            catkin_add_gtest(jacobian_test test/jacobian_test.cpp)
+            if(TARGET jacobian_test)
+                target_link_libraries(jacobian_test ${catkin_LIBRARIES} ${PROJECT_NAME})
+                target_include_directories(jacobian_test PRIVATE ${catkin_INCLUDE_DIRS})
+            endif()
+
+            catkin_add_gtest(hc_cc_circle_test test/hc_cc_circle_test.cpp)
+            if(TARGET hc_cc_circle_test)
+                target_link_libraries(hc_cc_circle_test ${catkin_LIBRARIES} ${PROJECT_NAME})
+                target_include_directories(hc_cc_circle_test PRIVATE ${catkin_INCLUDE_DIRS})
+            endif()
         endif()
+    elseif(${ROS_VERSION} EQUAL 2)
+        if(BUILD_TESTING)
+            find_package(ament_cmake_gtest)
+            ament_add_gtest(steering_functions_test test/steering_functions_test.cpp TIMEOUT 240)
+            target_link_libraries(steering_functions_test ${PROJECT_NAME})
 
-        catkin_add_gtest(fresnel_test test/fresnel_test.cpp)
-        if(TARGET fresnel_test)
-            target_link_libraries(fresnel_test ${catkin_LIBRARIES} ${PROJECT_NAME})
-            target_include_directories(fresnel_test PRIVATE ${catkin_INCLUDE_DIRS})
-        endif()
+            ament_add_gtest(fresnel_test test/fresnel_test.cpp)
+            target_link_libraries(fresnel_test ${PROJECT_NAME})
 
-        catkin_add_gtest(timing_test test/timing_test.cpp)
-        if(TARGET timing_test)
-            target_link_libraries(timing_test ${catkin_LIBRARIES} ${PROJECT_NAME})
-            target_include_directories(timing_test PRIVATE ${catkin_INCLUDE_DIRS})
-        endif()
+            # ament_add_gtest(timing_test test/timing_test.cpp)
```

Comment:
Why commented? What about the other tests?

Reply:
I've now also ported this test to ROS2 and kept it backwards compatible with ROS1. It makes it a bit more complicated, but I think its still quite readable.